packages/server/src/fhir/precommit.ts

```diff
@@ -25,22 +25,28 @@ export function isPreCommitSubscription(subscription: WithId<Subscription>): boo
 /**
  * Performs pre-commit validation for a resource by executing any associated pre-commit bots.
  * Throws an error if the bot execution fails.
+ * @param author - The author of the resource, used to check against blacklisted authors for pre-commit
  * @param project  - The project to which the resource belongs. Must be passed separately because this is called before the resource meta is assigned.
  * @param resource  - The resource to validate.
  * @param interaction - The interaction type (e.g., 'create', 'update', 'delete').
+ * @returns The validated resource if a pre-commit bot returns one, otherwise undefined.
  */
 export async function preCommitValidation(
+  author: Reference,
   project: WithId<Project> | undefined,
   resource: WithId<Resource>,
   interaction: BackgroundJobInteraction
-): Promise<void> {
+): Promise<Resource | boolean | undefined> {
```

Comment:
is returning `undefined` useful here? Should we just return `true` in these exit cases?